From acf29d06a7450ececf7ef4270973075aeb9ff845 Mon Sep 17 00:00:00 2001
From: Leonid Yegoshin <Leonid.Yegoshin@imgtec.com>
Date: Wed, 24 Apr 2013 12:22:31 -0500
Subject: [PATCH 095/105] MIPS: Malta: Remove ttyS2 serial.

Fix the following commit:

   commit 225ae5fd9a320e22841410049c3bdb6cf14a5841
   Author: Ralf Baechle <ralf@linux-mips.org>
   Date:   Tue Jan 22 12:59:30 2013 +0100

   MIPS: Malta: Fix interupt number of CBUS UART.

This leaves CBUS UART in the "MIPS" controller which effectively
switches off Status.IM4 during closure of ttyS2 and totally ruins
intercore communication in CMP environment. Just disable the CBUS
UART for now when CONFIG_MIPS_CMP is selected.

Signed-off-by: Leonid Yegoshin <Leonid.Yegoshin@imgtec.com>
Signed-off-by: Steven J. Hill <Steven.Hill@imgtec.com>
---
 arch/mips/mti-malta/malta-platform.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/mips/mti-malta/malta-platform.c b/arch/mips/mti-malta/malta-platform.c
index 132f866..e1dd1c1 100644
--- a/arch/mips/mti-malta/malta-platform.c
+++ b/arch/mips/mti-malta/malta-platform.c
@@ -47,6 +47,7 @@
 static struct plat_serial8250_port uart8250_data[] = {
 	SMC_PORT(0x3F8, 4),
 	SMC_PORT(0x2F8, 3),
+#ifndef CONFIG_MIPS_CMP
 	{
 		.mapbase	= 0x1f000900,	/* The CBUS UART */
 		.irq		= MIPS_CPU_IRQ_BASE + MIPSCPU_INT_MB2,
@@ -55,6 +56,7 @@ static struct plat_serial8250_port uart8250_data[] = {
 		.flags		= CBUS_UART_FLAGS,
 		.regshift	= 3,
 	},
+#endif
 	{ },
 };
 
-- 
1.7.1

